XForm/XForm/Data/XArray.cs

```diff
@@ -13,27 +13,34 @@ namespace XForm.Data
     ///  It provides indirection on the Array through ArraySelector, allowing filtering, 
     ///    re-ordering, and lookups without copying the raw array.
     ///  It provides null support around non-nullable types (which are faster to read and write)
-    ///    via an optional IsNull array.
+    ///    via an optional Nulls array.
     ///    
     ///  Usage:
     ///     T[] realArray = (T[])xarray.Array;                                                   // Array is of ColumnDetails.Type. Only one cast for the whole array.
     ///     for(int i = 0; i &lt; xarray.Count; ++i)                                             // Always loop from zero to xarray.Count - 1.
     ///     {
     ///         int realIndex = xarray.Index(i);                                                 // Index() is an inlined method which returns the real index of a row
-    ///         bool valueIsNull = (xarray.IsNull != null &amp;&amp; xarray.IsNull[realIndex]);   // IsNull, if provided, indicates whether the row is null
+    ///         bool valueIsNull = (xarray.HasNulls &amp;&amp; xarray.Nulls[realIndex]);   // IsNull, if provided, indicates whether the row is null
     ///         T rowValue = realArray[realIndex];
     ///     }
     /// </summary>
     public struct XArray
     {
-        private static bool[] s_NullSingle;
+        private static bool[] s_SingleTrue = new bool[1] { true };
+        private static bool[] s_SingleFalse = new bool[1] { false };
 
         /// <summary>
         ///  Optional array when XArray may contain null values indicating which
         ///  are null. If the array itself is null, none of the values are null.
         ///  Avoids using Nullable which keeps the values back-to-back for bulk serialization.
         /// </summary>
-        public bool[] IsNull { get; private set; }
+        public bool[] Nulls { get; private set; }
```

Comment:
Hmm. I like HasNulls, but I don't think we want to rename the IsNull array. That causes a lot of code changes everywhere.

Reply:
As a new user to the system, I believe the name 'IsNull' is very confusing to someone not familiar with the code base.  According to the C# Microsoft design guidelines, we should "name collection properties with a plural phrase describing the items in the collection instead of using a singular phrase followed by "List" or "Collection."  'IsNull' is named in such a way that fools users into thinking it's a single boolean field.  The new name of 'Nulls' conforms to the guidelines and also couples well with the new property 'HasNulls'.  

---
In reply to: [166131593](https://github.com/Microsoft/elfie-arriba/pull/176#discussion_r166131593) [](ancestors = 166131593)